### ISSUE TITLE
`setTimeout` does not execute

### ISSUE THREAD

**Issue by pladaria** [jhmaster2000/bun-repl#5]
```sh
$ bun-repl
Welcome to Bun.js v0.7.3
Type ".help" for more information.
> setTimeout(() => console.log('hello'), 1000)
Object [Timeout] {}
> 
```

I expect to see a "hello" printed, like in node:

```
$ node
Welcome to Node.js v18.12.1.
Type ".help" for more information.
> setTimeout(() => console.log('hello'), 1000)
Timeout {
[...]
}
> hello
```

**Comment by jhmaster2000** [comment]
This is actually just a small symptom of a much larger long standing known issue, it's caused by the synchronous input handling via FFI, which causes all types asynchronous work to not be able to tick inside the REPL (this includes timeouts, intervals, promises, etc), so it's not that it doesn't print but rather it never runs at all.

The only fix for this is completely switching to a proper async input handler method, which is planned as part of an in-progress major rewrite that will be released as soon as some bugs on Bun's side are resolved, which were the reason synchronous FFI input had to be used at the start to begin with.

TL;DR: Will be implicitly fixed on the next major release, ETA is this week/Bun 0.8 release.

**Comment by jhmaster2000** [comment]
This will be fixed on the 2.0.0 rewrite branch releasing shortly.

**Comment by not-my-profile** [comment]
I was also just very confused why `await Bun.write()` in `bun repl` did absolutely nothing.

If 2.0.0. will take some more time to land in `bun repl`, I think it would make sense to print some warning for top-level awaits in the meantime to avoid confusing users.

**Comment by jhmaster2000** [comment]
`bun repl` should always be going to the latest version, a lot of people seem to be running into a cache bug preventing that from happening though, try:

`rm -rf` the folder at the path given to you by running `bun pm cache` and then try re-running `bun repl`

**Comment by not-my-profile** [comment]
I installed bun for the first time today, so I don't think this is the issue. (I still tried it and it didn't change anything.)

Does `await Bun.write('/tmp/foo', 'test')` work for you in `bun repl`? For me it just prints `"use strict"` without creating the file.

**Comment by jhmaster2000** [comment]
Ah I see, that's a different issue that is indeed on the repl's side, I already have a fix for it locally, will be fixed shortly on the next release.

**Comment by jhmaster2000** [comment]
@not-my-profile the new release is out now, the REPL command should auto-update, could you try it again now?

**Comment by not-my-profile** [comment]
How does the auto update work? Nothing appears to have changed.

**Comment by jhmaster2000** [comment]
It works based on `bunx` currently, `bun repl` is an alias to `bunx bun-repl`, which should ideally always resolve to the latest.

Can you try running `.info` inside the REPL? If you **don't see** a line saying "REPL version" or if you do but it's not 2.0.5, then something is going wrong with the version resolution.
